src/components/Pool/PairsList/PairsList.tsx

```diff
@@ -0,0 +1,188 @@
+import { BigintIsh, CurrencyAmount, Pair, Percent, SingleSidedLiquidityMiningCampaign } from '@swapr/sdk'
+
+import React, { useEffect, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { Link } from 'react-router-dom'
+import { Box, Flex, Text } from 'rebass'
+import styled from 'styled-components'
+
+import { useActiveWeb3React } from '../../../hooks'
+import { useSWPRToken } from '../../../hooks/swpr/useSWPRToken'
+import { useIsMobileByMedia } from '../../../hooks/useIsMobileByMedia'
+import { useNativeCurrencyUSDPrice } from '../../../hooks/useNativeCurrencyUSDPrice'
+import { usePage } from '../../../hooks/usePage'
+import { useResponsiveItemsPerPage } from '../../../hooks/useResponsiveItemsPerPage'
+import { getStakedAmountUSD } from '../../../utils/liquidityMining'
+import { ButtonPrimary } from '../../Button'
+import { Pagination } from '../../Pagination'
+import { UndecoratedLink } from '../../UndercoratedLink'
+import { DimBlurBgBox } from '../DimBlurBgBox/styleds'
+import { PairsFilterType } from '../ListFilter'
+import { LoadingList } from './LoadingList'
+import { Pair as PairCard } from './Pair'
+
+export interface AggregatedPairs {
+  pair: Pair
+  liquidityUSD: CurrencyAmount
+  maximumApy: Percent
+  staked?: boolean
+  containsKpiToken?: boolean
+  hasFarming?: boolean
+  startsAt?: BigintIsh
+}
```

Comment:
It also used `useAllPairsWithLiquidityAndMaximumApyAndStakingIndicator` hook.

Don't know where to put it 🤔 

Should it have its own file?

Reply:
Well, good question. I think you can move it to `hook` it will be a little bit better. I know it's hard to organize code with current folder structure.